### PR TITLE
Fix MPP Task hang in spill random fail point

### LINKED ISSUE #7440: MPP Task hang in spill random fail point
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 2. What did you expect to see? (Required)

mpp task is released quickly.

### 3. What did you see instead (Required)

https://clinic.pingcap.com.cn/grafana/d/SVbh2xUWkoperatormaster/tiflash-summary-cluster-clinic?orgId=1&var-tidb_cluster=7227604775299467630&var-instance=All&var-store=All&from=1682807849000&to=1682813931000&var-tidb_version=master&var-org=31&refresh=15m

image

### PR DESCRIPTION
**Pull request by xzhangxian1008**
### What problem does this PR solve?

Issue Number: close #7440

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by windtalker — APPROVED** [review]
LGTM

File: dbms/src/Flash/Mpp/GRPCReceiveQueue.h [hunk 1]
```diff
@@ -71,10 +71,14 @@ class AsyncRequestHandlerWaitQueue
         return wait_retry_queue.empty();
     }
 
-    void push(AsyncRetryConnection conn)
+    // Return false when AsyncRequestHandlerWaitQueue is closed
+    bool push(AsyncRetryConnection conn)
     {
         std::lock_guard lock(mu);
+        if (is_closed)
```
Comment (SeaRise) [inline dbms/src/Flash/Mpp/GRPCReceiveQueue.h:78]:
```suggestion
        if unlikely (is_closed)
```

**Review by SeaRise — APPROVED** [review]
Others LGTM

**Pull request merged**